contracts/base/ERC721Permit.sol

```diff
@@ -46,23 +45,18 @@ abstract contract ERC721Permit is ERC721, IERC721Permit {
         0x49ecf333e5b8c95c40fdafc95c1ad136e8914a8fb55e9dc8bb01eaa83a2df9ad;
```

Comment:
I prefer saving the constant as keccak256(string..) 

also can we move all this to a hashing library? `PermitHash.hash(spender, tokenId, nonce, deadline)`?

And then we can build another library to verify the signature after the hash

so permit code looks something like:

```
function permit() checkDeadline(deadline) {

_useNonce()


bytes32 permitHash = PermitHash.hash(spender, tokenId, deadline, nonce);
signature.verify(hash, owner); // handles sc signatures as well

_approve(operator) 

}


```